open-api/rest-catalog-open-api.yaml

```diff
@@ -3405,6 +3490,101 @@ components:
       allOf:
         - $ref: '#/components/schemas/ScanTasks'
 
+    GetEventsRequest:
+      type: object
+      properties:
+        page-token:
+          $ref: "#/components/schemas/PageToken"
+        page-size:
+          type: integer
+          format: int32
+          description: >
+            The maximum number of events to return in a single response.
+            If not provided, the server may choose a default page size.
+            Servers may return less results than requested for various reasons, such as
+            server side limits, payload size or processing time.
+        after-timestamp-ms:
+          type: integer
+          format: int64
+          description: >
+            The (server) timestamp in milliseconds to start consuming events from (inclusive).
+            If not provided, the first available timestamp is used.
+        operation-types:
+          type: array
+          items:
+            $ref: "#/components/schemas/OperationType"
+          description: >
+            Filter events by the type of operation.
+            If not provided, all types are returned.
+        catalog-objects:
+          type: array
+          discriminator:
+            propertyName: reference-type
+            mapping:
+              namespace: "#/components/schemas/NamespaceReference"
+              table: "#/components/schemas/TableReference"
+              view: "#/components/schemas/ViewReference"
+          items:
+            oneOf:
+              - $ref: "#/components/schemas/NamespaceReference"
+              - $ref: "#/components/schemas/TableReference"
+              - $ref: "#/components/schemas/ViewReference"
+          description: >
+            List of catalog objects (namespaces, tables, views) to get events for.
+            If not provided, events for all objects will be returned subject to other filters.
+            For specified namespaces, events for the namespaces and all containing objects
+            (namespaces, tables, views) will be returned.
+        custom-filters:
```

Comment:
Why have a sub-object of custom filters rather than setting `additionalProperties: true` on the request object?

I think this also allows passing any JSON structure. Do we want to limit that as we do elsewhere?

```yaml
  additionalProperties:
    type: string
```

Reply:
I prefer to separate custom filters to avoid potential collisions with filters that we standardize later.

I added `type: string`

Reply:
A filter on actor for example